src/main/java/de/tum/in/www1/artemis/web/rest/RatingResource.java

```diff
@@ -0,0 +1,120 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.forbidden;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import de.tum.in.www1.artemis.domain.Rating;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.ParticipationService;
+import de.tum.in.www1.artemis.service.RatingService;
+import de.tum.in.www1.artemis.service.ResultService;
+import de.tum.in.www1.artemis.service.UserService;
+
+/**
+ * REST controller for managing Rating.
+ */
+@RestController
+@RequestMapping("/api")
+public class RatingResource {
+
+    private static final String ENTITY_NAME = "rating";
+
+    private final Logger log = LoggerFactory.getLogger(RatingResource.class);
+
+    private final RatingService ratingService;
+
+    private final UserService userService;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final ResultService resultService;
+
+    private final ParticipationService participationService;
+
+    public RatingResource(RatingService ratingService, UserService userService, AuthorizationCheckService authCheckService, ResultService resultService,
+            ParticipationService participationService) {
+        this.ratingService = ratingService;
+        this.userService = userService;
+        this.authCheckService = authCheckService;
+        this.resultService = resultService;
+        this.participationService = participationService;
+    }
+
+    /**
+     * Return Rating referencing resultId or 404 Not Found
```

Comment:
I think we should avoid raising a 404 simply because a result doesn't have a rating yet. I think a 404 should mostly be reserved for actual unexpected not found errors since it will also end up in Sentry and the error statistics. Having a 404 raised every time someone looks at an assessed submission without result rating would generate a lot of 404s.